File: app/schemas/notification.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class NotificationOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    type: str
    title: str
    message: str
    related_token: Optional[str] = None
    is_read: bool
    created_at: datetime

File: frontend/src/types/notification.ts
export type NotificationType = 'coding_test' | 'interview' | 'status_update' | 'application_confirmation'

export interface CandidateNotification {
  id: number
  type: NotificationType
  title: string
  message: string
  related_token: string | null
  is_read: boolean
  created_at: string
}
